src/Microsoft.AspNet.Razor.Runtime/TagHelpers/TagHelperAttribute.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.Internal.Web.Utils;
+
+namespace Microsoft.AspNet.Razor.Runtime.TagHelpers
+{
+    public class TagHelperAttribute : IReadOnlyTagHelperAttribute
```

Comment:
Ultimately I feel its odd to move the IEquatable up for a sub-class. As for loss of functionality here's an ex:

```C#
var allAttrs = new ReadOnlyTagHelperAttributes<MyTagHelperAttribute>(existingAttrs);
var attr = new MyTagHelperAttribute(....);
allAttrs.IndexOf(attr); // Doesn't use the comparer if TagHelperAttribute is the IEquatable class since we implemented IReadOnlyTagHelperAttribute in our MyTagHelperAttribute class
```